File: 1356-sort-integers-by-the-number-of-1-bits/1356-sort-integers-by-the-number-of-1-bits.py
class Solution:
    def sortByBits(self, arr: List[int]) -> List[int]:
        arr.sort()
        n = len(arr)
        for i in range(n):
            num = arr[i]
            t = 0
            while num!=0:
                num = num & (num-1)
                t  += 1
            
            arr[i]=[arr[i],t]
        
        arr.sort(key=lambda y:y[1])
        ans=[]
        for i in arr:
            ans.append(i[0])
        return ans

File: 1356-sort-integers-by-the-number-of-1-bits/1356-sort-integers-by-the-number-of-1-bits.cpp
#define vti vector<int>
#define pb push_back
#define ppb pop_back

class Solution {
public:
    static int countOnes(int a)
    {
        int c = 0;
        while(a)
        {
            a = a&(a-1);
            c++;
        }
        return c;
    }
    static bool compare(int&a, int& b)
    {
        // return __builtin_popcount(a) < __builtin_popcount(b);
        int p = countOnes(a);
        int q = countOnes(b);
        
        if(p == q)
            return a<=b;
        return p<q;
    }
    vti sortByBits(vector<int>& arr)
    {
        vti res = arr;
        sort(res.begin(), res.end(), compare);
        
        return res;
    }
};